### PR TITLE
Unit test PatientInvoiceFormService

### PR DESCRIPTION
**Pull request by jniles**
This PR adds unit tests using Karma to the PatientInvoiceForm.  The tests are not complete, but major steps have been tested, including the initialization step, setting a patient, and digests. It also rewrites the PatientInvoiceFormService to comply with ESLint.  Finally, a new service has been added to Karma to allow sharing mocked data between tests called `bhima.mocks`.

### CODE REVIEW

File: test/client-unit/services/PatientInvoiceForm.spec.js [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+    const subsidies = Mocks.subsidies();
+
+    expect(form.prices.data).to.have.deep.members(priceList.items);
+
+    // test that all ids of invoice fees are in the form.
+    const invoiceFeeIds = form.invoicingFees.map(fee => fee.id);
+
+    // console.error(form.invoicingFees);
+    // console.error(' === ');
+    // console.error(invoiceFees);
```
Comment (mbayopanda) [inline test/client-unit/services/PatientInvoiceForm.spec.js]: these comments are still here

File: client/src/modules/invoices/patientInvoiceForm.service.js [hunk 2]
```diff
@@ -465,15 +456,15 @@ Patients, PriceLists, Inventory, AppCache, Store, Pool,
...
-      var item = this.addItem();
+    cp.items.forEach((cacheItem) => {
+      const item = this.addItem();
 
       item.inventory_uuid = cacheItem.inventory_uuid;
       this.configureItem(item);
 
       item.quantity = cacheItem.quantity;
       item.transaction_price = cacheItem.transaction_price;
-    }.bind(this));
```
Comment (mbayopanda) [inline client/src/modules/invoices/patientInvoiceForm.service.js:476]: By removing this `.bind(this)`, the function inside the foreach can still see the parent `this` and call it like ``this.addItem(); ?

**Review by mbayopanda — APPROVED** [review]
LGTM
I run the test locally on my machine it works fine.
👍

File: client/src/modules/invoices/patientInvoiceForm.service.js (hunk 2 above)
Comment (jniles) [reply client/src/modules/invoices/patientInvoiceForm.service.js:476]: Yup!  That is the power of arrow functions in JavaScript.  [Here is the MDN documentation on it](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions#No_separate_this)

File: test/client-unit/services/PatientInvoiceForm.spec.js (hunk 1 above)
Comment (jniles) [reply test/client-unit/services/PatientInvoiceForm.spec.js]: Whoops, you are right. :(

**Comment by jniles** [comment]
@mbayopanda changes have landed!

**Pull request merged**